### PR TITLE
Added Prime Factor [C] 

### PR DESCRIPTION
**Pull request by yashLadha**
Closes https://github.com/iiitv/algos/issues/26

#### By submitting this pull request I confirm I've read and complied with the below requirements.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm name} [{Language}]`, not `Update README.md` or `Added new code`.

### CODE REVIEW

**Comment by aviaryan** [comment]
Also the function should take array pointer as input and return the list of factors in the output. The return should be the size of the factors array. What do you think @singhpratyush  @yashLadha ? 

```C
int primeFactors(int * retArray, int n){

// calculate for n
    return arSize;    
}

**Comment by yashLadha** [comment]
Sir Is array should contain primes ?

**Comment by aviaryan** [comment]
@yashLadha Please make yourself more clear.

**Comment by yashLadha** [comment]
The array which will be passed in the function, that array should contain prime upto some range, or what?

File: prime_factor/prime_factor.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#include <stdio.h>
+
+int prime_factor(unsigned int prime[1000], int no, int len) {
```
Comment (jsroyal) [inline prime_factor/prime_factor.c]: use pointer instead of a simple array pass.

File: prime_factor/prime_factor.c [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    for (int i = 2; i < 60000 && *ptr < 1000; ++i) {
+        int fl = 0;
+        for (int j = 2; j < i; ++j) {
+            if (i % j == 0) {
+                fl = 1;
+                break;
+            }
+        }
+        if (fl == 0) {
+            ar[*ptr] = i;
```
Comment (singhpratyush) [inline prime_factor/prime_factor.c]: Storing `int` to `unsigned int`.

File: prime_factor/prime_factor.c [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    int idx = 0; // last index of the prime number
+    prime_generator(primes, &idx);
+
+    int cnt = 0; // the no. of prime factor of no
+    // prime factors of no
+    printf("Prime factors: ");
+    for (int i = 0; i < idx && primes[i] <= no; ++i) {
+        if (no % primes[i] == 0) {
+            cnt++;
+            printf("%d ", primes[i]);
```
Comment (singhpratyush) [inline prime_factor/prime_factor.c]: `primes[i]` is `unsigned int`, but printed using `%d`. No prints except `main` functions.

**Pull request closed without merging**